src/write/line.rs

```diff
@@ -632,6 +647,24 @@ impl LineProgram {
                 if self.file_has_md5 {
                     w.write(&info.md5)?;
                 }
+                match (self.file_has_embedded_source, info.embedded_source.as_ref()) {
+                    (true, Some(embedded_source @ LineString::LineStringRef(..))) => {
+                        embedded_source.write(
+                            w,
+                            constants::DW_FORM_line_strp,
+                            self.encoding,
+                            debug_line_str_offsets,
+                            debug_str_offsets,
+                        )?;
+                    }
+                    (true, Some(ref val)) => {
+                        panic!("Can only write LineString::LineStringRef for embedded sources, got {:?}", val);
+                    }
+                    (true, None) => {
+                        panic!("has_embedded_source is true, but no source set");
```

Comment:
this is highly suboptimal. I am thinking of just writing a empty string reference here.